hack/sdk/gen-sdk.sh

```diff
@@ -38,6 +39,9 @@ go install k8s.io/kube-openapi/cmd/openapi-gen@${OPENAPI_VERSION}
 echo "Generating OpenAPI specification ..."
 ${GOPATH}/bin/openapi-gen --input-dirs github.com/fluid-cloudnative/fluid/api/v1alpha1 --output-package github.com/fluid-cloudnative/fluid/api/v1alpha1 --go-header-file hack/boilerplate.go.txt
 
+echo "Downloading codegen jar ..."
```

Comment:
I suggest improving the flexibility and robustness of the code.

1. Before downloading the swagger-codegen-cli.jar file, check if it already exists locally. If it exists, do not attempt to download it again.

2. If the download fails, add a retry mechanism to avoid failures due to network fluctuations.

The modified code adds more robustness and reliability to the script by checking for the presence of curl, avoiding unnecessary downloads of existing files, and retrying failed downloads.


```bash
echo "Downloading codegen jar ..."
if ! command -v curl >/dev/null 2>&1; then
    echo "Error: curl command not found." >&2
    exit 1
fi

if [ -f "${SWAGGER_CODEGEN_JAR}" ]; then
    echo "Using existing ${SWAGGER_CODEGEN_JAR}"
else
    for i in {1..3}; do
        if curl -fLsS "${SWAGGER_CODEGEN_JAR_URL}" -o "${SWAGGER_CODEGEN_JAR}"; then
            break
        elif [ "$i" -eq 3 ]; then
            echo "Failed to download ${SWAGGER_CODEGEN_JAR} after 3 attempts." >&2
            exit 1
        else
            echo "Failed to download ${SWAGGER_CODEGEN_JAR}, retrying in 10 seconds..." >&2
            sleep 10
        fi
    done
fi
```

Reply:
Thanks for your suggestion and code! I think the curl command should only be checked if swagger-codegen-cli.jar file doesn't exist locally.